### PR TITLE
New xstate proposal

### PR DESCRIPTION
**Pull request by thisispatrick33**

### CODE REVIEW

File: components/SearchField.vue [hunk 1]
```diff
@@ -29,21 +30,24 @@ import SuggestionsListItem from '@/components/SuggestionsListItem.vue'
...
+  routeMachine.value.state.matches('appointmentPage.location.locationSelected')
 )
 
-const showSuggestionsList = computed(() =>
-  routeMachine.value.state.matches('appointmentPage.location.showSuggestions')
+const showSuggestionsList = computed(
+  () =>
+    routeMachine.value.state.matches(
+      'appointmentPage.suggestionsList.suggestionsListOpened'
+    ) && routeMachine.value.state.context.suggestions.length > 0
```
Comment (peter9ke) [inline components/SearchField.vue:50]: here should be also specific xstate state

File: machines/routeMachine.ts [hunk 2]
```diff
@@ -79,5 +81,15 @@ export const routeMachine = createMachine(
       ...routeMachineActions,
       ...appointmentPageActions,
     },
+    services: {
+      fetchLocations: (context) => {
+        if (context.location) {
+          return fetchLocations(context.location)
+        } else {
+          return fetchLocations(context.locationSearchField)
+        }
```
Comment (peter9ke) [inline machines/routeMachine.ts:90]: context.location, we need call

Comment (jakubednar) [reply machines/routeMachine.ts:90]: rename to locationInUrl

File: machines/actions/appointmentPageActions.ts [hunk 3]
```diff
@@ -1,27 +1,30 @@
...
 import { sortByDistance } from '@/utils/sortByDistance'
 import { resources } from '@/data/resources'
+import { normalizeTimeslots } from '@/utils/appointments'
+import { Context } from '@/machines/routeMachine'
 
-const setUrlParams = (params: UrlParams) => {
+const setUndefinedIfValueIsNull = (
+  param: string | null
+): string | undefined => {
+  return param === null ? undefined : param
```
Comment (peter9ke) [inline machines/actions/appointmentPageActions.ts:10]: What is the point of this?

File: machines/routeMachine.ts (hunk 2 above)
Comment (peter9ke) [reply machines/routeMachine.ts:93]: this is still open

**Pull request merged**